lib/puppet/functions/archive/assemble_nexus_url.rb

```diff
@@ -0,0 +1,15 @@
+Puppet::Functions.create_function(:'archive::assemble_nexus_url') do
+  require "cgi"
+
+  dispatch :assemble_nexus_url do
+    required_param 'String', :nexus_url
+    required_param 'Hash', :params
+  end
+
+  def assemble_nexus_url(nexus_url, params)
+    service_relative_url = 'service/local/artifact/maven/content'
```

Comment:
Should this be a param with a default too?

Reply:
I don't it should.